woocommerce/woocommerce-gutenberg-products-block
# Add Product Search Results Template

This PR adds a new template called `Product Search Results`. This template will be rendered only on the block themes when the user lands on `?s={keyword}&post_type=product`. 




Fixes #6218

### Screenshots

![image](https://user-images.githubusercontent.com/4463174/162420290-524b17fb-63b7-41fe-8538-a3107c7f7be3.png)
*Product Search Results in the template list*

![image](https://user-images.githubusercontent.com/4463174/162420366-083f4fae-aae5-4f3a-957c-dde267da112f.png)
*Editor view when the user edits the Product Search Result template*

![image](https://user-images.githubusercontent.com/4463174/162420494-afc59411-a6a2-4c95-8590-7b919719f51d.png)
*Frontend side when the user lands on `?s=shirt&post_type=product` url. The template Product Search Results is rendered*


### Testing

#### Automated Tests
### Manual Testing

How to test the changes in this Pull Request:

Check out this branch

1. Be sure that you are using a block theme (as `Twenty Twenty-Two`).
2. Append this `?s={product_to_search}&post_type=product` to your website URL. For example `http://dev.local/?s=shirt&post_type=product`. You should see a grid with the products that match the keyword (or a not found message if any product matches the keyword). Keep this page open.
3. On the Dashboard, click on Appearance->Editor.
4. On the Site Editor, click on the WordPress Logo on the top left and after that click on Templates.
5. Be sure that the Search Result Template is visible. If yes, click on it.
6. Add new blocks and save the template.
7. Refresh the page that you opened in step 2. Be sure that the customizations are reflected on the frontend side. 
8. Back on the templates list and restore the template. Be sure that the customizations are gone.
9. Enable a classic theme (as Storefront). Check that if you visit `?s={product_to_search}&post_type=product` everything works well. **IMPORTANT:** Remember that for the classic themes the templates are not loaded. So the implementation of this page depends on the theme. 

#### Extendibility for the themes
With these steps, we will check that the themes can extend the new template:

1. Be sure that you are using a block theme (as `Twenty Twenty-Two`).
2. Open the folder related to the active block theme in your website folder. For `Twenty Twenty-Two` the path is `/themes/twentytwentytwo/`.
3. Open the folder `templates` and create a file `search-results-product.html` with [this content](https://gist.github.com/gigitux/dd13ed87db5c6ca5601eeac1e9830ab8).
4. Append this `?s={product_to_search}&post_type=product` to your website URL. For example `http://dev.local/?s=shirt&post_type=product`. You should see a grid with the products that match the keyword  (or a not found message if any product matches the keyword) and a paragraph with the text `Column One, Paragraph One`.
5. On the Dashboard, click on Appearance->Editor.
6. On the Site Editor, click on the WordPress Logo on the top left and after that click on Templates.
7.  Be sure that the Product Search Results template is visible and added by the theme.
8. Edit the Product Search Results template.
10. Be sure that the customizations are reflected on the frontend side. 
11. Enable a classic theme (as `Storefront`). Check that if you visit `?s={product_to_search}&post_type=product` everything works well. **IMPORTANT:** Remember that for the classic themes the templates are not loaded. So the implementation of this page depends on the theme. 


### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).

1. Be sure that you are using a block theme (as `Twenty Twenty-Two`).
2. Append this `?s={product_to_search}&post_type=product` to your website URL. For example `http://dev.local/?s=shirt&post_type=product`. You should see a grid with the products that match the keyword (or a not found message if any product matches the keyword). Keep this page open.
3. On the Dashboard, click on Appearance->Editor
4. On the Site Editor, click on the WordPress Logo on the top left and after that click on Templates.
5. Be sure that the Search Result Template is visible. If yes, click on it.
6. Add new blocks and save the template.
7. Refresh the page that you opened in step 2. 
8. Back on the templates list and restore the template.
9. Enable a classic theme (as `Storefront`). Check that if you visit `?s={product_to_search}&post_type=product` everything works well. **IMPORTANT:** Remember that for the classic themes the templates are not loaded. So the implementation of this page depends on the theme. 

### Changelog

> Add Product Search Results template.
